build.sbt

```diff
@@ -1,23 +1,62 @@
 import xerial.sbt.Sonatype._
+import Dependencies._
 
-cancelable in Global := true
-onLoad in Global ~= (_ andThen ("project core" :: _))
-
-lazy val credentialSettings = Seq(
-  credentials += Credentials(
-    "GnuPG Key ID",
-    "gpg",
-    System.getenv().get("GPG_KEY_ID"),
-    "ignored"
+lazy val commonSettings = Seq(
+  // We are overriding the default behavior of sbt-git which, by default, only
+  // appends the `-SNAPSHOT` suffix if there are uncommitted changes in the
+  // workspace.
+  version := {
+    if (git.gitDescribedVersion.value.isEmpty)
+      git.gitHeadCommit.value.get.substring(0, 7) + "-SNAPSHOT"
```

Comment:
`gitDescribedVersion` [uses](https://github.com/sbt/sbt-git/blob/3dd1343ba8c2d77cda92080f65aab91331822a1a/src/main/scala/com/typesafe/sbt/git/ReadableGit.scala#L18) the version as returned by `git describe --tags`. 

This build doesn't have any tags, so `git.gitDescribedVersion.value.get` was null:

```bash
$ git describe --tags
fatal: No names found, cannot describe anything.
```

An alternative approach would be to source the version from `git describe --always` which will show an abbreviated SHA as a fallback. This isn't yet possible (https://github.com/sbt/sbt-git/issues/155), although there is a [hacky solution](https://github.com/sbt/sbt-git/issues/161#issuecomment-469342173) that overrides the value of `gitDescribedVersion`.

I think this solution works just as well, but I'd like a critical eye since this deviates from the code block we've now copied across multiple repositories. We haven't run into an issue before because we've always introduced publishing into a project that already has tags.

Reply:
I think this is fine.

The way it is set up now though, once [this](https://github.com/azavea/stac4s/pull/3/files#diff-1d37e48f9ceff6d8030570cd36286a61R59) is reverted the publish step should only be running for tags I think - though I see how that could make testing difficult. 

We can take the current master branch and tag it as `0.1.0` if you would like.

Reply:
Cool. 

I think it's OK to leave this change in because it fixes undesirable behavior. Even if the code path isn't likely to be exercised, it is likely we'll use this project as a model in the future.